src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -475,14 +481,15 @@ private TypeSpec CreateEnumerableSpec(TypeParseInfo typeParseInfo)
                     {
                         populationCastType = CollectionPopulationCastType.NotApplicable;
                     }
-                    else if (_typeSymbols.GenericICollection is not null && GetInterface(type, _typeSymbols.GenericICollection_Unbound) is not null)
+                    else if (isCollection)
                     {
                         // implements ICollection<> -- cast to it
                         populationCastType = CollectionPopulationCastType.ICollection;
                     }
                     else
                     {
-                        return CreateUnsupportedCollectionSpec(typeParseInfo);
+                        // not a collection
+                        return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=24&length=12)

Can't this type is implementing a non-generic ICollection?

Reply:
Same as above.  It could, but we don't treat those as bindable collections.
https://github.com/dotnet/runtime/blob/2106bb0c57c7bc39c9b5320668c09321651b99e6/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L455-L459